jaxrs-tck/src/main/java/jakarta/ws/rs/tck/signaturetest/jaxrs/JAXRSSigTestIT.java

```diff
@@ -0,0 +1,342 @@
+   * result of this method call.
+   *
+   * @param sigArgsList
+   *          The arg list being constructed to pass to the utility that records
+   *          and runs signature file tests.
+   */
+  private static void addNoContainerPkgs(List<String> sigArgsList) {
+    for (int i = 0; i < NO_CONTAINER_PKGS.length; i++) {
+      sigArgsList.add(NO_CONTAINER_PKGS[i]);
+    }
+  }
+
+  /**
+   * Returns a list of strings where each string represents a package name. Each
+   * package name will have it's signature tested by the signature test
+   * framework.
+   * 
+   * @param vehicleName
+   *          The name of the Jaspic container where the signature tests should
+   *          be conducted.
+   * @return String[] The names of the packages whose signatures should be
+   *         verified.
+   */
+  protected String[] getPackages(String vehicleName) {
+    List<String> packages = new LinkedList<String>();
+
+    if (vehicleName.equals(NO_VEHICLE)) {
+      addNoContainerPkgs(packages);
+    } else {
+      addDefaultPkgs(packages); // add default vehicle packages
+      if (vehicleName.equals(EJB_VEHICLE) || vehicleName.equals(SERVLET_VEHICLE)
+          || vehicleName.equals(JSP_VEHICLE)) {
+        addEjbServletJspPkgs(packages);
+      }
+      if (vehicleName.equals(SERVLET_VEHICLE)
+          || vehicleName.equals(JSP_VEHICLE)) {
+        addServletJspPkgs(packages);
+      }
+    }
+    return packages.toArray(new String[packages.size()]);
+  }
+
+  /***** Boilerplate Code *****/
+
+
+  /*
+   * The following comments are specified in the base class that defines the
+   * signature tests. This is done so the test finders will find the right class
+   * to run. The implementation of these methods is inherited from the super
+   * class which is part of the signature test framework.
+   */
+
+  // NOTE: If the API under test is not part of your testing runtime
+  // environment, you may use the property sigTestClasspath to specify
+  // where the API under test lives. This should almost never be used.
+  // Normally the API under test should be specified in the classpath
+  // of the VM running the signature tests. Use either the first
+  // comment or the one below it depending on which properties your
+  // signature tests need. Please do not use both comments.
+
+  /*
+   * @class.setup_props: ts_home, The base path of this TCK; sigTestClasspath;
+   */
+  /*
+   * @testName: signatureTest
+   * 
+   * @assertion: A JAXRS container must implement the required classes and APIs
+   * specified in the JAXRS Specification.
+   * 
+   * @test_Strategy: Using reflection, gather the implementation specific
+   * classes and APIs. Compare these results with the expected (required)
+   * classes and APIs.
+   *
+   */
+  @Test
+  public void signatureTest() throws Fault {
+    TestUtil.logMsg("$$$ JAXRSSigTestIT.signatureTest() called");
+    SigTestResult results = null;
+    String mapFile = System.getProperty("signature.mapfile");
```

Comment:
It is the tradeoff that we had to do for having the convenience of sigtest-maven-plugin jar as a maven dependency. We invoke the `run` method within the plugin jar that only accepts relative path of above files as arguments as in the file jaxrs-tck/src/main/java/jakarta/ws/rs/tck/signaturetest/SigTestDriver.java (method `runSignatureTest`()). Since these files are present _within the jar_ it had to be extracted and placed at an accessible location. 

In jakartaee-tck we had the above files at paths relative to the ts_home hence it was not necessary.

Reply:
Thanks for the suggestion to create temp files in java.io.tmpdir directory. 
Created temporary files for signature.mapfile & signature.packagelist using File.createTempFile and signature file using File.createNewFile. signature.repositoryDir will be java.io.tmpdir where the signature file is created.